nifs/atomvm_adc.c

```diff
@@ -432,51 +395,50 @@ static term nif_adc_take_reading(Context *ctx, int argc, term argv[])
         if (UNLIKELY(memory_ensure_free(ctx, 3) != MEMORY_GC_OK)) {
             RAISE_ERROR(OUT_OF_MEMORY_ATOM);
         } else {
-            term error_tuple = term_alloc_tuple(2, ctx);
+            term error_tuple = term_alloc_tuple(2, &ctx->heap);
             term_put_tuple_element(error_tuple, 0, ERROR_ATOM);
-            term_put_tuple_element(error_tuple, 1, context_make_atom(ctx, invalid_pin_atom));
+            term_put_tuple_element(error_tuple, 1, globalcontext_make_atom(ctx->global, invalid_pin_atom));
             return error_tuple;
         }
     }
 
     term read_options = argv[1];
     VALIDATE_VALUE(read_options, term_is_list);
     term samples = interop_proplist_get_value_default(
-        read_options, context_make_atom(ctx, samples_atom), term_from_int(DEFAULT_SAMPLES)
-    );
+        read_options, globalcontext_make_atom(ctx->global, samples_atom), term_from_int(DEFAULT_SAMPLES));
```

Comment:
We can use `interop_kv_get_value_default` without having to make an atom.  Let's use that here.